### PR TITLE
add OCR-D/ocrd_fileformat, fix #80

### LINKED ISSUE #80: Add ocrd_fileformat

### PR DESCRIPTION
**Pull request by kba**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -474,9 +481,9 @@ endif
 docker-minimum-git docker-medium-git docker-maximum-git: PIP_OPTIONS = -e
 
 # Minimum-size selection: use Ocropy binarization, use Tesseract from PPA
-docker-minimum docker-minimum-git: DOCKER_MODULES = core ocrd_im6convert ocrd_cis ocrd_tesserocr tesserocr workflow-configuration ocrd_repair_inconsistencies
+docker-minimum docker-minimum-git: DOCKER_MODULES = core ocrd_im6convert ocrd_cis ocrd_tesserocr tesserocr workflow-configuration ocrd_repair_inconsistencies ocrd_fileformat
```
Comment (stweil) [inline Makefile:484]: I am friend of alphabetic sorting of such lists, but that can be done next time.

File: .gitmodules [hunk 2]
```diff
@@ -70,3 +70,6 @@
 [submodule "ocrd_repair_inconsistencies"]
 	path = ocrd_repair_inconsistencies
 	url = https://github.com/qurator-spk/ocrd_repair_inconsistencies.git
+[submodule "ocrd_fileformat"]
+	path = ocrd_fileformat
+	url = https://github.com/OCR-D/ocrd_fileformat
```
Comment (stweil) [inline .gitmodules:75]:
```suggestion
	url = https://github.com/OCR-D/ocrd_fileformat.git
```

**Comment by bertsky** [comment]
LGTM.

Can you please add ocrd-pagetopdf as well ([this version](https://github.com/JKamlah/ocrd_pagetopdf/commit/467c8919ee7860609f4c33229dfcd52e5fc2f23b) for now)? I know it should be in ocr-fileformat itself, but that would never be able to replace the full-featured direct conversion.

File: Makefile (hunk 1 above)
Comment (kba) [reply Makefile]: https://github.com/OCR-D/ocrd_all/pull/83/commits/238f5147925d6ccf0b70ede1d892556f3a4b6a5a `core` should be before `cor-asv-...` IMHO

**Comment by kba** [comment]
https://app.circleci.com/pipelines/github/OCR-D/ocrd_all/119/workflows/3520cde8-ea4f-4eb8-8e8e-ab229cb8991a/jobs/127 Looks like the submodule is not checked out recursively - @bertsky @stweil what am I doing wrong?

> Can you please add ocrd-pagetopdf as well (this version for now)? I know it should be in ocr-fileformat itself, but that would never be able to replace the full-featured direct conversion.

Sure.

**Comment by bertsky** [comment]
> https://app.circleci.com/pipelines/github/OCR-D/ocrd_all/119/workflows/3520cde8-ea4f-4eb8-8e8e-ab229cb8991a/jobs/127 Looks like the submodule is not checked out recursively - @bertsky @stweil what am I doing wrong?

You made no mistake: `ocrd_fileformat` gets cloned recursively:
```
git submodule sync --recursive ocrd_fileformat
if git submodule status --recursive ocrd_fileformat | grep -qv '^ '; then \
	git submodule update --init --recursive ocrd_fileformat && \
	touch ocrd_fileformat; fi
Submodule 'ocrd_fileformat' (https://github.com/OCR-D/ocrd_fileformat) registered for path 'ocrd_fileformat'
Cloning into '/build/ocrd_fileformat'...
Submodule path 'ocrd_fileformat': checked out 'de5b53282243c0b9696bf1ca9a22c327793ef37d'
Submodule 'ocr-fileformat' (https://github.com/UB-Mannheim/ocr-fileformat) registered for path 'ocrd_fileformat/ocr-fileformat'
Cloning into '/build/ocrd_fileformat/ocr-fileformat'...
Submodule path 'ocrd_fileformat/ocr-fileformat': checked out '86a41edf80ad235a95a612b981ffb68ae49417db'
```

But then subsequently:
```
. /usr/local/bin/activate && make -C ocrd_fileformat install-fileformat install
make[1]: Entering directory '/build/ocrd_fileformat'
make -C  ocr-fileformat PREFIX=/usr/local vendor install
make[2]: Entering directory '/build/ocrd_fileformat/ocr-fileformat'
make -C vendor check
make[3]: Entering directory '/build/ocrd_fileformat/ocr-fileformat/vendor'
make[3]: *** [check] Error 1
make[2]: *** [check] Error 2
make[1]: *** [install-fileformat] Error 2
```

So obviously there's a bug in ocr-fileformat: `make vendor` does not work, and does not even report a proper error.

**Comment by bertsky** [comment]
> > Can you please add ocrd-pagetopdf as well (this version for now)? I know it should be in ocr-fileformat itself, but that would never be able to replace the full-featured direct conversion.
> 
> Sure see #84

Thanks @stweil @kba!

**Comment by stweil** [comment]
> So obviously there's a bug in ocr-fileformat: make vendor does not work, and does not even report a proper error.

It checks whether all required executables are available, and the error message is indeed not good. Maybe `unzip` is missing.

**Comment by stweil** [comment]
@kba, @bertsky, would you prefer a different build fix, or can we merge now?

**Comment by bertsky** [comment]
> would you prefer a different build fix, or can we merge now?

I suppose we could give ocrd_fileformat's Makefile a target `deps-ubuntu` with all required packages. But we could also do that in a later PR and see how we fare without.

**Pull request merged**